app/src/main/java/com/duckduckgo/app/browser/WebViewRequestInterceptor.kt

```diff
@@ -48,6 +49,7 @@ interface RequestInterceptor {
         request: WebResourceRequest,
         webView: WebView,
         documentUri: Uri?,
+        maliciousSiteProtectionWebViewIntegration: MaliciousSiteBlockerWebViewIntegration,
```

Comment:
Is there a specific reason for passing this as a method argument rather than injecting it as a class attribute via the constructor? I would prefer the later since this is a dependency rather than a value.

Reply:
Singleton + state is something I would be really careful about, race conditions are coming. It's not only users moving to another tab, we also have some scenarios where 2 tabs could be loading a site, and I expect this also changing in the project of swipe tabs. 
For transparency, I would add `shouldOverrideUrlLoading` method to `RequestInterceptor`, only with the logic that applies here, that will simplify things. I know you don't like, and you feel we should then move all the logic we have on that method, I don't think that's necessary.

I'm going to leave you my suggestion if you want to keep the singleton, let's remove the state then (processedUrls). That will save us from unexpected scenarios. If we think is necessary, we can revisit.